lib/ctf-writer/writer.c

```diff
@@ -161,22 +164,20 @@ struct bt_ctf_writer *bt_ctf_writer_create(const char *path)
 		goto error_destroy;
 	}
 
-	writer->trace_dir_fd = open(path, O_RDONLY, S_IRWXU | S_IRWXG);
-	if (writer->trace_dir_fd < 0) {
-		perror("open");
-		goto error_destroy;
-	}
-
-	writer->metadata_fd = openat(writer->trace_dir_fd, "metadata",
+	writer->metadata_fd = open(metadata_path,
 		O_WRONLY | O_CREAT | O_TRUNC,
 		S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
 
+	// jgalar: Should we handle open() failure here?
+
+	g_free(metadata_path);
 	return writer;
 
 error_destroy:
-	unlinkat(writer->trace_dir_fd, "metadata", 0);
+	unlink(metadata_path);
```

Comment:
Should only be done after the `open` done on the metadata file.

In fact, that line is not needed as the creation of the metadata file is the last action taken. In other words, if an error occurs, we won't have created it anyway.